pg/src/main/java/org/bouncycastle/openpgp/PGPKeyRingGenerator.java

```diff
@@ -99,7 +99,7 @@ public PGPKeyRingGenerator(
             hashedVec.add(existing.packets[i]);
         }
         this.hashedPcks = new PGPSignatureSubpacketVector(
-            (SignatureSubpacket[])hashedVec.toArray(new SignatureSubpacket[hashedVec.size()]));
+                hashedVec.toArray(new SignatureSubpacket[0]));
```

Comment:
Here we can omit `hashedVec.size()`: https://stackoverflow.com/a/29444594

Reply:
Yes, interesting... I don't know, most of the PGP API is the same classes right down to Java 1. It is heavily used on Java 4 and Java 3 platforms (the new "j2me") as well. We filter out the generics for Java 4 and below, but the "meaningless" casts are still required for the other profiles based on the older JVMs. I'm not really sure which variant serves the greater good in this case.

Reply:
the array initialization is just a micro-optimization anyways.

>but the "meaningless" casts are still required for the other profiles based on the older JVMs

That's interesting, I did not know that.

Reply:
Yes, it's a very broad church around here. I'm working my way through looking for public methods we've missed.